src/plonk/prover.rs

```diff
@@ -20,256 +20,399 @@ use crate::transcript::TranscriptWrite;
+            drop(aux_commitments_projective);
+            metrics::counter!("aux_commitments", aux_commitments.len() as u64);
+
+            for commitment in &aux_commitments {
+                transcript
+                    .common_point(*commitment)
+                    .map_err(|_| Error::TranscriptError)?;
+            }
+
+            let aux_polys: Vec<_> = aux
+                .iter()
+                .map(|poly| {
+                    let lagrange_vec = domain.lagrange_from_vec(poly.to_vec());
+                    domain.lagrange_to_coeff(lagrange_vec)
+                })
+                .collect();
 
-    let aux_cosets: Vec<_> = meta
-        .aux_queries
-        .iter()
-        .map(|&(column, at)| {
-            let poly = aux_polys[column.index()].clone();
-            domain.coeff_to_extended(poly, at)
+            let aux_cosets: Vec<_> = meta
+                .aux_queries
+                .iter()
+                .map(|&(column, at)| {
+                    let poly = aux_polys[column.index()].clone();
+                    domain.coeff_to_extended(poly, at)
+                })
+                .collect();
+
+            Ok(AuxSingle {
+                aux_values: aux.to_vec(),
+                aux_polys,
+                aux_cosets,
+            })
         })
         .collect();
 
-    // Compute commitments to advice column polynomials
-    let advice_blinds: Vec<_> = witness
-        .advice
-        .iter()
-        .map(|_| Blind(C::Scalar::rand()))
-        .collect();
-    let advice_commitments_projective: Vec<_> = witness
-        .advice
-        .iter()
-        .zip(advice_blinds.iter())
-        .map(|(poly, blind)| params.commit_lagrange(poly, *blind))
-        .collect();
-    let mut advice_commitments = vec![C::zero(); advice_commitments_projective.len()];
-    C::Projective::batch_to_affine(&advice_commitments_projective, &mut advice_commitments);
-    let advice_commitments = advice_commitments;
-    drop(advice_commitments_projective);
-    metrics::counter!("advice_commitments", advice_commitments.len() as u64);
+    let aux_vec = match aux_vec {
+        Ok(aux_vec) => aux_vec,
+        Err(err) => return Err(err),
+    };
 
-    for commitment in &advice_commitments {
-        transcript
-            .write_point(*commitment)
-            .map_err(|_| Error::TranscriptError)?;
+    struct AdviceSingle<C: CurveAffine> {
+        pub advice_values: Vec<Polynomial<C::Scalar, LagrangeCoeff>>,
+        pub advice_polys: Vec<Polynomial<C::Scalar, Coeff>>,
+        pub advice_cosets: Vec<Polynomial<C::Scalar, ExtendedLagrangeCoeff>>,
+        pub advice_blinds: Vec<Blind<C::Scalar>>,
     }
 
-    let advice_polys: Vec<_> = witness
-        .advice
-        .clone()
-        .into_iter()
-        .map(|poly| domain.lagrange_to_coeff(poly))
-        .collect();
-
-    let advice_cosets: Vec<_> = meta
-        .advice_queries
+    let advice_vec: Result<Vec<_>, _> = circuits
```

Comment:
```suggestion
    let advice_vec: Vec<AdviceSingle<C>> = circuits
```
It'd be helpful for legibility to fully specify these result types.